signer/src/transaction_coordinator.rs

```diff
@@ -1577,11 +1600,217 @@ where
         })
     }
 
+    /// Fetches pending withdrawal requests from storage and filters them based
+    /// on the remaining consensus rules as defined in #741:
+    ///
+    /// 1. [x] The request must not have been swept within the current canonical
+    ///    Bitcoin chain.
+    /// 2. [x] The request must be confirmed in a canonical Stacks block.
+    /// 3. [x] The request must have reached the required number of Bitcoin
+    /// 4. [x] The request must be approved:
+    ///     - [x] By the required number of signers (this is implemented as a
+    ///       pre-filter in the query, any signer),
+    ///     - [x] And by the required number of signers _in the current signer
+    ///       set_.
+    /// 5. [ ] The request has been approved by this signer (this shouldn't be
+    ///    applicable to the coordinator).
+    /// 6. [ ] The assessed fees will be within the constraints of the request's
+    ///    specified maximum fee (this is handled during packaging).
+    /// 7. [x] The request must not have expired (handled in the query).
+    /// 8. [x] The request amount must be above the dust limit.
+    /// 9. [x] The request must be within the current sBTC caps.
+    #[tracing::instrument(skip_all)]
+    pub async fn get_eligible_pending_withdrawal_requests<DB>(
+        storage: &DB,
+        withdrawal_blocks_expiry: u16,
+        params: &GetPendingRequestsParams<'_>,
+    ) -> Result<Vec<utxo::WithdrawalRequest>, Error>
+    where
+        DB: DbRead,
+    {
+        // Constants used for logging (local to this method).
+        const REQUEST_SKIPPED_MESSAGE: &str = "skipping withdrawal request";
+        const SKIP_REASON_AMOUNT_IS_DUST: &str = "amount_is_dust";
+        const SKIP_REASON_PER_WITHDRAWAL_CAP_EXCEEDED: &str = "per_withdrawal_cap_exceeded";
+        const SKIP_REASON_INSUFFICIENT_CONFIRMATIONS: &str = "insufficient_confirmations";
+        const SKIP_REASON_INSUFFICIENT_VOTES: &str = "insufficient_votes";
+
+        let mut eligible_withdrawals = Vec::new();
+
+        // We set the context window for the pending-accepted query below to the
+        // number of blocks that the withdrawal request is considered
+        // valid (not expired). This limits the number of blocks the query will
+        // consider when fetching pending withdrawal requests.
+        let context_window = withdrawal_blocks_expiry;
+
+        // Fetch pending withdrawal requests from storage. This method performs
+        // the following filtering according to consensus rules:
```

Comment:
What I am suggesting needs some work, but it is closer to being true. What is written here suggests that the function makes sure that the returned requests are not expired. Maybe try something like this?
```suggestion
        // Fetch pending withdrawal requests from storage. This method, with
        // the given inputs, performs the following filtering according to
        // consensus rules:
```

Reply:
Tweaked it a bit in [97a9f1a](https://github.com/stacks-network/sbtc/pull/1399/commits/97a9f1ab42e59919a0aabaf7c628e2c5d3c96136), thoughts?

Reply:
👍🏽